### PR TITLE
[Vue] convert sites manager single site template/controller to vue component

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of https://github.com/matomo-org/matomo/pull/18553.**

Changes:
* Fix typing error in SiteSelector.
* Convert sites-manager-site.controller.js to SiteFields.vue component.
* Add stores for timezone/currency/site type for use in SitesManager.
* Created lazyInitSingleton function that constructs a singleton when it is first accessed so we don't end up sending a bunch of ajax requests on page load for stores that may not be used.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
Here a few issues found while clicking through the UI. Some might possibly be unrelated to the changes here.
Only did some quick testing yet. Maybe something else pops up when I continue the review...

- [ ] When adding a new site the cancel button does not work
- [ ] When searching for a site has exactly 10 results the `next` button is clickable (even though there aren't any further results)

**Comment by diosmosis** [comment]
> When searching for a site has exactly 10 results the next button is clickable (even though there aren't any further results)

This is a problem w/ 4.x-dev. It can be fixed easily but better to wait for these fixes until the next PR since the fix would still be in the angularjs code here.

**Comment by diosmosis** [comment]
> When adding a new site the cancel button does not work

Fixed.

**Comment by sgiehl** [comment]
@diosmosis Adding a rollup doesn't work with the changes here. The field for selecting the sites isn't displayed without any error. It works on 4.x-dev, so seems to be related to the changes here. Can you check if we can get that running again without changes in the RollUp plugin?

**Comment by diosmosis** [comment]
@sgiehl should be fixed

File: plugins/SitesManager/vue/src/SiteFields/SiteFields.vue [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+      let suffix = '';
+      if (this.isInternalSetupUrl) {
+        suffix = MatomoUrl.stringify({
+          idSite: site.idsite,
+          period: MatomoUrl.parsed.value.period,
+          date: MatomoUrl.parsed.value.date,
+          updated: 'false',
+        });
+      }
+      return `${this.howToSetupUrl}${suffix}`;
```
Comment (sgiehl) [inline plugins/SitesManager/vue/src/SiteFields/SiteFields.vue]: Here it seems a `&` is missing between the url and suffix

**Review by sgiehl — CHANGES_REQUESTED** [review]
Found some more issues:

- When adding a new site, it automatically added at the top of the list. But the currency and timezone is always empty for those new sites:
  image
  When editing a site the display values in the list are also not updated.

Besides that it looks fine and should be good to merge, once those issues are fixed.

**Comment by diosmosis** [comment]
> When adding a new site, it automatically added at the top of the list. But the currency and timezone is always empty for those new sites:

This is broken on 4.x-dev too. It should be fixed in the next PR.

> Here it seems a & is missing between the url and suffix

Fixed.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**